manager/apis/app/v1alpha1/condition_types.go

```diff
@@ -0,0 +1,32 @@
+// Copyright 2020 IBM Corp.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
+// ConditionType represents a condition type
+type ConditionType string
+
+// Constants defining condition types
+const (
+	ErrorCondition ConditionType = "Error"
+	DenyCondition  ConditionType = "Deny"
+	ReadyCondition ConditionType = "Ready"
+)
+
+// Condition describes the state of a FybrikApplication at a certain point.
+type Condition struct {
+	// Type of the condition
+	Type ConditionType `json:"type"`
+	// Status of the condition: true or false
```

Comment:
Should we add something like:
// +kubebuilder:validation:Optional
// +kubebuilder:validation:Enum=True,False,Unknown
 // +kubebuilder:default:=Unknown

Reply:
Added